src/FileConventions/Library.fs

```diff
@@ -104,3 +104,100 @@ let EolAtEof(fileInfo: FileInfo) =
                 False
         else
             True
+
+let IsFooterReference(line: string) : bool =
+    line.[0] = '[' && line.IndexOf "] " > 0
+
+let IsFixesOrClosesSentence(line: string) : bool =
+    line.IndexOf "Fixes " = 0 || line.IndexOf "Closes " = 0
+
+let IsCoAuthoredByTag(line: string) : bool =
+    line.IndexOf "Co-authored-by: " = 0
+
+let IsFooterNote(line: string) : bool =
+    IsFooterReference line
+    || IsCoAuthoredByTag line
+    || IsFixesOrClosesSentence line
+
+type Word =
+    | CodeBlock
+    | FooterNote
+    | PlainText
+
+type Text(textStr: string, textType: Word) =
```

Comment:
@realmarv there's no reason for why this type cannot be a record instead of a class (records are preferred in F#, classes should be seldomly used, because records are idiomatic F#), unless I'm missing something

Reply:
Done